home/components/Profile.js

```diff
@@ -284,16 +297,34 @@ export default class Profile extends React.Component {
       },
       async buttonIndex => {
         const option = options[buttonIndex];
+
+        if (option === 'Cancel') return;
+
+        const { status } = await Permissions.askAsync(Permissions.CAMERA, Permissions.CAMERA_ROLL);
+        if (status !== Permissions.PermissionStatus.GRANTED) {
+          alert(
+            'Cannot select a banner photo without media access! Please enable the "Camera" & "Camera Roll" permission in your system settings.'
+          );
+          return;
+        }
+
+        const mediaOptions = {
+          allowsEditing: true,
+          quality: 1.0,
+          allowsMultipleSelection: false,
+          mediaTypes: ImagePicker.MediaTypeOptions.Images,
+          exif: false,
+          base64: false,
+        };
+
+        let media;
         if (option === 'Take Photo') {
-          this.props.navigation.navigate('Camera', {
-            username: this.props.username,
-            belongsToCurrentUser: this.props.isOwnProfile,
-          });
+          media = await ImagePicker.launchCameraAsync(mediaOptions);
         } else if (option === 'Choose Photo') {
-          this.props.navigation.navigate('MediaLibrary', {
-            username: this.props.username,
-            belongsToCurrentUser: this.props.isOwnProfile,
-          });
+          media = await ImagePicker.launchImageLibraryAsync(mediaOptions);
+        }
+        if (!media.cancelled) {
+          Store.dispatch(ProfileActions.setImage(media.uri));
```

Comment:
Move the image (probably to the caches directory so it doesn't take up space forever) so that if it's deleted from the camera roll it doesn't disappear right away. (Or if ImagePicker already moves images to the caches directory, just use the given URI.)